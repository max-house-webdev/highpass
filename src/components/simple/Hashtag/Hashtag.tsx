import { Text, TextProps } from "@chakra-ui/react";

export interface IHashtagProps extends TextProps {
  textContent: string;
}

export function Hashtag(props: IHashtagProps) {
  const { textContent, ...rest } = props;

  return <Text variant={"hashtag"} {...rest}>{`#${textContent}`}</Text>;
}
