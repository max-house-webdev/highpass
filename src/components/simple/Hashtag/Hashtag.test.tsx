import React from "react";

import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { Hashtag } from "./Hashtag";

describe("<Hashtag />", () => {
  test("it should mount", () => {
    render(<Hashtag textContent={"hashtag"} />);

    const hashtag = screen.getByTestId("hashtag");

    expect(hashtag).toBeInTheDocument();
    expect(hashtag).toHaveTextContent(/#\w/);
  });
});

test("<LearnMore /> renders correctly", () => {
  const tree = renderer.create(<Hashtag textContent={"hashtag"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
