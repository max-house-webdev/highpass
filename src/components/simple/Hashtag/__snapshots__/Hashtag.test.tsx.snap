// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LearnMore /> renders correctly 1`] = `
<p
  className="chakra-text css-0"
  data-testid="hashtag"
>
  #hashtag
</p>
`;
